utilities/mas-imu-test/src/MasImuTest.cpp

```diff
@@ -0,0 +1,1361 @@
+        return false;
+    }
+
+    if (m_period < 0)
+    {
+        yError() << "[MasImuTest::configure] The period cannot be negative. Configuration failed.";
+        return false;
+    }
+
+    ok = m_parametersPtr->getParameter("robot", m_commonDataPtr->robotName);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
+        return false;
+    }
+    std::string robotModelName;
+    ok = m_parametersPtr->getParameter("model", robotModelName);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
+        return false;
+    }
+    std::string pathToModel = yarp::os::ResourceFinder::getResourceFinderSingleton().findFileByName(robotModelName);
+    yInfo() << "[MasImuTest::configure] Robot model path: " << pathToModel;
+    iDynTree::ModelLoader modelLoader;
+    if (!modelLoader.loadModelFromFile(pathToModel))
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
+        return false;
+    }
+
+    m_commonDataPtr->fullModel = modelLoader.model();
+
+    std::string baseLink;
+    ok = m_parametersPtr->getParameter("base_link", baseLink);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
+        return false;
+    }
+
+    iDynTree::LinkIndex baseLinkIndex = m_commonDataPtr->fullModel.getLinkIndex(baseLink);
+    if (baseLinkIndex == iDynTree::LINK_INVALID_INDEX)
+    {
+        yError() << "[MasImuTest::configure] The link " << baseLink << " does not exists in " << robotModelName
+                 << ". Configuration failed.";
+        return false;
+    }
+
+    ok = m_commonDataPtr->fullModel.computeFullTreeTraversal(m_commonDataPtr->traversal, baseLinkIndex);
+
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Failed to build the traversal. Configuration failed.";
+        return false;
+    }
+
+    iDynTree::Rotation baseRotation;
+    if(!iDynTree::parseRotationMatrix(rf, "base_rotation", baseRotation))
+    {
+        baseRotation = iDynTree::Rotation::Identity();
+        yInfo() << "[MasImuTest::configure] Using the identity as desired rotation for the additional frame";
+    }
+
+    ok = iDynTree::isValidRotationMatrix(baseRotation);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] The specified base rotation is not a rotation matrix.";
+        return false;
+    }
+
+    m_commonDataPtr->baseTransform = iDynTree::Transform::Identity();
+    m_commonDataPtr->baseTransform.setRotation(baseRotation);
+
+    ok = m_parametersPtr->getParameter("filter_yaw", m_commonDataPtr->filterYaw);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
+        return false;
+    }
+
+    double minJointVariationInDeg;
+    ok = m_parametersPtr->getParameter("min_joint_variation_deg", minJointVariationInDeg);
+    if (!ok)
+    {
+        yError() << "[MasImuTest::configure] Configuration failed.";
```

Comment:
Maybe adding due to which variable configuration failed could be helpful while running the tests.

Reply:
Right, I modified them in https://github.com/dic-iit/bipedal-locomotion-framework/pull/62/commits/f053b43fec94159eaf02696cd3e69a659cf9feb9. In any case, the ``ParametersHandler`` throws an error when the parameter is not found.